src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -855,13 +855,23 @@ void MFEMSidreDataCollection::Load(const std::string& path,
   DataCollection::DeleteAll();
   // Reset DataStore?
 
+  int numInputRanks = 1;
+  int numCommRanks = 1;
+
   #if defined(AXOM_USE_MPI) && defined(MFEM_USE_MPI)
   if(m_comm != MPI_COMM_NULL)
   {
-    IOManager reader(m_comm);
     // The conduit abstraction appears to automatically handle the ".root"
     // suffix, but the IOManager does not, so it gets added here
-    reader.read(m_bp_grp->getDataStore()->getRoot(), path + ".root");
+    using axom::utilities::string::endsWith;
+    std::string suffixedPath = endsWith(path, ".root") ? path : path + ".root";
+
+    IOManager reader(m_comm);
+    reader.read(m_bp_grp->getDataStore()->getRoot(), suffixedPath);
+
+    // Get some data in support of error checks below
+    numInputRanks = reader.getNumGroupsFromRoot(suffixedPath);
+    MPI_Comm_size(m_comm, &numCommRanks);
```

Comment:
DC bugfix/enhancement 2: Improved logging (a few lines down), if loading on a different number of ranks than were used for save.
Note @nselliott: this required making a private function from `IOManager` public (`getNumGroupsFromRoot()`)